testsuite/tests/workflows/init-with-pin/test.py

```diff
@@ -0,0 +1,67 @@
+"""
+Test a basic init-with-pin-run workflow.
+
+TODO: this test relies on external resources (GitHub repositories for the
+hello/libhello crates sources). This makes this test slower than needed and
+potentially unstable. We should refactor it to rely only on testsuite files
+instead.
+"""
+
+import os
+
+from drivers.alr import run_alr
+from drivers.asserts import assert_eq
+
+
+def check_line_in(filename, line):
+    """
+    Assert that the `filename` tetx file contains at least one line that
+    contains `line`.
+    """
+    with open(filename, 'r') as f:
+        for l in f:
+            if l.rstrip() == line:
+                break
+        else:
+            assert False, 'Could not find {} in {}'.format(
+                repr(line), filename)
+
+
+# Create a new "xxx" program project
+run_alr('init', '--bin', 'xxx')
+os.chdir('xxx')
+
+# Make it depend on libhello
+session_file = os.path.join('alire', 'xxx.toml')
+run_alr('with', 'libhello')
+check_line_in(session_file, 'libhello = "any"')
+
+# Add the corresponding "with" line in xxx.gpr.
+#
+# TODO: maybe "alr with" should do that automatically?
```

Comment:
This is something I have considered in the past. For releases with a single GPR it seems the obvious thing to do. For ones with several... I guess we shouldn't (or we might ask the user interactively?)

As con, I don't like to touch user files directly. Again, we could ask beforehand (incidentally, we have -n to avoid interactivity -- which currently only happens before doing a `sudo apt-get install`)

Reply:
Good points. Anyway, tt would be nice to have but clearly not needed :)